test/e2e/upgrade_test.go

```diff
@@ -0,0 +1,131 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	nc "github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachine/nodeconfig"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"log"
+)
+
+func upgradeTestSuite(t *testing.T) {
+
+	t.Run("Node deletion during upgrade", func(t *testing.T) { testUpgradeVersion(t) })
+	t.Run("Node deletion during annotation tampering", func(t *testing.T) { testTamperAnnotation(t) })
+}
+
+// testUpgradeVersion tests if the operator deletes machines and recreats them, if the node annotation doesn`t match
+// with the expected annotation
+func testUpgradeVersion(t *testing.T) {
+	testCtx, err := NewTestContext(t)
+	require.NoError(t, err)
+
+	operatorVersion, err := testCtx.getWMCOVersion()
+	require.NoError(t, err, "could not get WMCO version")
+
+	err = testCtx.waitForWindowsNodesAnnotation(gc.numberOfNodes, operatorVersion)
+	if err != nil {
+		t.Fatalf("windows node upgrade failed  with %v", err)
```

Comment:
Correct unless you want to hard fail here.

Reply:
I think we should hard fail here. The `err` implies that on version change, the nodes are not recreated and configured as expected. So the further tests would fail anyway

Reply:
[Fatal](https://golang.org/pkg/testing/#T.Fatal) is just a logging wrapper around `FailNow()`. This is the same as [require](https://github.com/stretchr/testify/blob/master/require/require.go#L1199) 

The two are functionally identical but with require we get cleaner code :)

If you want to prevent future tests from running if this test fails, you should restructure the tests making the dependent tests subtests.

cc @aravindhp